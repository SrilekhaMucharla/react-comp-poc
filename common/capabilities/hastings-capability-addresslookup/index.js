// eslint-disable-next-line import/prefer-default-export
export { default as HastingsAddressLookupService } from './services/HastingsAddressLookupService';
