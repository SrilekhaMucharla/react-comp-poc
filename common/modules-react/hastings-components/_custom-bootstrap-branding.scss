// Do nothing here

//TODO - move these definitions

///* import only the necessary Bootstrap files */
//@import "../../../node_modules/bootstrap/scss/functions";
//@import "../../../node_modules/bootstrap/scss/variables";
//@import "../../../node_modules/bootstrap/scss/mixins";
//
//@import "./custom-bootstrap-branding-variables";
//
///* import bootstrap to set changes */
//@import "../../../node_modules/bootstrap/scss/bootstrap";

//body {
//  font-family: $font-family-sans-serif;
//}


//.justify-content-md-center {
//  justify-content: center !important;
//}

//.btn-group-toggle {
//  display: unset;
//
//  > .btn:not(:last-child),
//  > .btn-group:not(:last-child) {
//    margin-right: 16px;
//  }
//
//  @include media-breakpoint-up(lg) {
//    > .btn:not(:last-child),
//    > .btn-group:not(:last-child) {
//      margin-right: 22px;
//    }
//  }
//
//  > .btn:not(:last-child):not(.dropdown-toggle),
//  > .btn-group:not(:last-child) > .btn {
//    @include border-radius(0.625rem);
//  }
//
//  > .btn:not(:first-child),
//  > .btn-group:not(:first-child) > .btn {
//    @include border-radius(0.625rem);
//  }
//
//
//  @include media-breakpoint-up(lg) {
//    > .btn {
//      width: 189px;
//      height: 60px;
//    }
//  }
//
//  > .btn {
//    display: inline-flex;
//    font-weight: 600;
//    font-size: 1.125rem;
//    margin-bottom: 30px;
//
//    > .justify-content-md-center {
//      align-items: center;
//      flex-direction: column;
//      display: flex;
//      width: 100%;
//    }
//  }
//
//  > .btn-primary:hover {
//    border-color: unset;
//  }
//}

//.horizontal-line {
//  width: 100%;
//  height: 1px;
//  border-radius: 6px;
//  background-color: $blue-transparent-30
//}