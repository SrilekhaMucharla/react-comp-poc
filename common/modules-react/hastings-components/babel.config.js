module.exports = require('./../../../common/build-tools/gw-build-config-babel');
