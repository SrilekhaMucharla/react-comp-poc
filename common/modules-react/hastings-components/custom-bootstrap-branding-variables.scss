/* make the customizations */
$theme-colors: (
        "primary": #fff,
);

$border-radius: 0.625rem;
$font-family-sans-serif: 'TTNormsPro-Bold', "Helvetica Neue", BlinkMacSystemFont, "Segoe UI", Roboto, Arial, "Noto Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";

$border-color: #727272;
$border-color-active: #585757;
$border-solid-xl: 1px solid $border-color;
$border-solid-active-button: 1px solid $border-color-active;
$background-color-active: #DDDDDD;
$secondaryButtonBGColor: #FFFFFF;
$headerTextColor: #333333;
$primaryButtonBGColor: #404040;
$tertiaryButtonBGColor: #4F4F4F;
$secondaryTextColor: $secondaryButtonBGColor;
$regularButtonBorderColor: #9C9C9C;

/** Custom Hastings colors, based on theme.scss from main App **/
$alternateBG: #f7f7f9;
$appTheme: #011831;
$pageBG: #ffffff;
$regNumberBackgroundColor: #FBE700;

$genericTitleBoldFontFace: $font-family-sans-serif;
$genericFontFace: 'TTNormsPro';
$genericNormalFontFace: 'TTNormsPro-Normal';
$genericRegularFontFace: 'TTNormsPro-Regular';
$fontStyleNormal: normal;
$sectionHeaderColor: #02213e;
$secondaryLink: #AFBDD1;

$commonCompoBGColor: #01355a;
$commonCompoBGColorLightBlue2: #f3f8ff;
$commonCompoBGColorBlue: #00448c;
$commonCompoBGColorRed:#a90d30;
$commonCompoBGColorWhite: #ffffff;
$commonCompoBGColorBlueLight:#0069cc;
$commonCompoBGColorGreen:#0e838a;
$commonCompoTextColorBlueDark : #011831;
$commonCompoTextColorBlueGray: #60759e;
$commonCompoTextColorBlueLight:#0069cc;
$commonCompoTextColorBlueGrayLight: #afbdd1;
$commonCompoTextColorGraySecondary:#4c6172;
$commonCompoTextColor: #ffffff;
$commonCompoBorderColor:  #60759e;
$commonCompoBorderColorRed:#e10930;
$commonCompoBorderColorGreen:#00a39e;
$commonCompoBorderColorGreenDark:#0e838a;
$commonCompoBorderColorBlueLight: #0085ff;
$commonCompoBorderColorBlueGrayLight: #afbdd1;

$customizeQuoteBtnColor: #0E838A;
$customizeQuoteSecondaryBtnColor: rgba(14, 131, 138, 0.2);
$customizeQuoteSecondaryBtnBorderColor: #00A39E;
$customizeQuotePopupBackgroundColor: rgba(1, 24, 49, 0.95);

$purple-secondary: #a24593;

$breakpoint-mobile: 375px;
$breakpoint-tablet: 768px;
$breakpoint-desktop: 1140px;

/* Exclusively for usage in styled components */
:export {
  appTheme: $appTheme;
  secondaryTextColor: $secondaryTextColor;
  commonCompoBGColor: $commonCompoBGColor;
  commonCompoBGColorWhite: $commonCompoBGColorWhite;
  commonCompoBGColorLightBlue2: $commonCompoBGColorLightBlue2;
  commonCompoTextColor: $commonCompoTextColor;
  commonCompoTextColorBlueDark: $commonCompoTextColorBlueDark;
  tertiaryTextColor: $commonCompoTextColorBlueGrayLight;
  genericFontFace: $genericFontFace;
  genericTitleBoldFontFace: $genericTitleBoldFontFace;
}