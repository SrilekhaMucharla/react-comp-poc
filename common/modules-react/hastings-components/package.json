{
  "name": "hastings-components",
  "version": "1.0.0",
  "description": "Hastings React Components Library for QnB",
  "main": "index.js",
  "author": "Hastings",
  "private": true,
  "dependencies": {
    "@babel/core": "^7.11.6",
    "@babel/preset-env": "^7.11.5",
    "babel-jest": "^26.3.0",
    "bootstrap": "^4.5.2",
    "formik": "^2.1.7",
    "react-bootstrap": "^1.3.0",
    "react-datepicker": "2.0.0",
    "react-input-mask": "^2.0.4",
    "react-intl": "^2.9.0",
    "react-lottie": "^1.2.3",
    "react-select": "^2.4.4",
    "react-test-renderer": "^16.13.1",
    "styled-components": "^5.2.0",
    "yup": "^0.29.3"
  },
  "peerDependencies": {
    "react": "16.x",
    "react-dom": "16.x",
    "prop-types": "15.x",
    "lodash": "4.x",
    "classnames": "^3.x"
  },
  "devDependencies": {
    "@babel/polyfill": "^7.12.1",
    "@babel/preset-react": "^7.10.4",
    "enzyme": "^3.11.0",
    "enzyme-adapter-react-16": "^1.15.5",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^26.4.2"
  },
  "scripts": {
    "test": "node ./../../build-tools/gw-build-testing/bin/runTest.js",
    "test:oneFile": "node ./../../build-tools/gw-build-testing/bin/runTest.js --oneFile",
    "test:watch": "jest --watchAll"
  }
}
