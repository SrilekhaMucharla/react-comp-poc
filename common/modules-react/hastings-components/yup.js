import _ from 'lodash';
import {
    mixed,
    string,
    number,
    bool,
    boolean,
    date,
    object,
    array,
    ref,
    lazy,
    reach,
    isSchema,
    addMethod,
    setLocale,
    ValidationError
} from 'yup';

const minOrEqual = function (compereTo, message) {
    return this.test('minOrEqual', message,
        (value) => {
            return new Date(value).setHours(0, 0, 0, 0) >= new Date(compereTo).setHours(0, 0, 0, 0);
        });
};
const before17thBirthday = function (message) {
    const maxDate = new Date();
    const minDate = new Date();
    maxDate.setFullYear(maxDate.getFullYear() - 17);
    maxDate.setDate(maxDate.getDate() + 30);
    return this.test('before17thBirthday', message,
        (value) => {
            const dateToCompare = new Date(value).setHours(0, 0, 0, 0);
            return !(dateToCompare <= new Date(minDate).setHours(0, 0, 0, 0)
             && dateToCompare >= new Date(maxDate).setHours(0, 0, 0, 0));
        });
};

const notInTheFuture = function (message) {
    return this.test('notInTheFuture', message,
        (value) => {
            const now = new Date().setHours(0, 0, 0, 0);
            if (value) {
                return now > new Date(value).setHours(0, 0, 0, 0);
            }
            return false;
        });
};

const VMValidation = function (vmPath, customMessage, submissionVM) {
    const defaultMessage = customMessage || 'Please enter valid value';
    this.vmPath = vmPath;
    return this.test('VMValidation', defaultMessage,
        function () {
            const { path, createError } = this;
            const vmValidationMessage = _.get(submissionVM, `${vmPath}.aspects.validationMessage`);
            return _.get(submissionVM, `${vmPath}.aspects.subtreeValid`) || createError({ path, message: vmValidationMessage || defaultMessage });
        });
};

// eslint-disable-next-line func-names
const notMobileNumber = function (message) {
    return this.test('notMobileNumber', message, (value) => !(value && value.length === 10));
};

// eslint-disable-next-line func-names
const mobileNumber = function (message) {
    return this.test('mobileNumber', message, (value) => value && value.startsWith('0'));
};

addMethod(mixed, 'VMValidation', VMValidation);
addMethod(date, 'minOrEqual', minOrEqual);
addMethod(date, 'before17thBirthday', before17thBirthday);
addMethod(date, 'notInTheFuture', notInTheFuture);
addMethod(string, 'notMobileNumber', notMobileNumber);
addMethod(string, 'mobileNumber', mobileNumber);

export {
    mixed,
    string,
    number,
    bool,
    boolean,
    date,
    object,
    array,
    ref,
    lazy,
    reach,
    isSchema,
    addMethod,
    setLocale,
    ValidationError,
    VMValidation
};
