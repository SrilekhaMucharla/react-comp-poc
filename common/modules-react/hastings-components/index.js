import HDInteractiveCard from './components/HDInteractiveCard/HDInteractiveCard';
import HDInteractiveCardRefactor from './components/HDInteractiveCardRefactor/HDInteractiveCardRefactor';
import HDTextInput from './components/HDTextInput/HDTextInput';
import HDTextInputRefactor from './components/HDTextInputRefactor/HDTextInputRefactor';
import HDButton from './components/HDButton/HDButton';
import HDButtonRefactor from './components/HDButtonRefactor/HDButtonRefactor';
import HDInfoCard from './components/HDInfoCard/HDInfoCard';
import HDInfoCardRefactor from './components/HDInfoCardRefactor/HDInfoCardRefactor';
import HDQuoteInfo from './components/HDQuoteInfo/HDQuoteInfo';
import HDQuoteInfoRefactor from './components/HDQuoteInfoRefactor/HDQuoteInfoRefactor';
import HDQuoteInfoWarning from './components/HDQuoteInfoWarning/HDQuoteInfoWarning';
import HDLabel from './components/HDLabel/HDLabel';
import HDLabelRefactor from './components/HDLabelRefactor/HDLabelRefactor';
import HDStepper from './components/HDStepper/HDStepper';
import HDToggleButtonGroup from './components/HDToggleButtonGroup/HDToggleButtonGroup';
import HDToggleButtonGroupRefactor from './components/HDToggleButtonGroupRefactor/HDToggleButtonGroupRefactor';
import HDClickableCard from './components/HDClickableCard/HDClickableCard';
import HDDropdownList from './components/HDDropdownList/HDDropdownList';
import HDImageRadioButton from './components/HDImageRadioButton/HDImageRadioButton';
import HDButtonDashed from './components/HDButtonDashed/HDButtonDashed';
import HDDataCard from './components/HDDataCard/HDDataCard';
import HDForm from './components/HDForm/HDForm';
import HDWarningModal from './components/HDWarningModal/HDWarningModal';
import HDOverlayPopup from './components/HDOverlayPopup/HDOverlayPopup';
import HDModal from './components/HDModal/HDModal';
import HDAsyncSelect from './components/HDDropdownList/HDAsyncSelect';
import HDInformationToolTip from './components/HDInformationToolTip/HDInformationToolTip';
import HDAlert from './components/HDAlert/HDAlert';
import HDAlertRefactor from './components/HDAlertRefactor/HDAlertRefactor';
import HDDatePicker from './components/HDDatePicker/HDDatePicker';
import HDDatePickerRefactor from './components/HDDatePickerRefactor/HDDatePickerRefactor';
import HDTable from './components/HDTable/HDTable';
import HDAccordion from './components/HDAccordion/HDAccordion';
import HDAccordionRefactor from './components/HDAccordionRefactor/HDAccordionRefactor';
import HDQuoteTable from './components/HDQuoteTable/HDQuoteTable';
import HDQuoteDownload from './components/HDQuoteDownload/HDQuoteDownload';
import HDQuoteDownloadRefactor from './components/HDQuoteDownloadRefactor/HDQuoteDownloadRefactor';
import HDPlaceholderWithHeader from './components/HDPlaceholderWithHeader/HDPlaceholderWithHeader';
import HDContainer from './components/HDContainer/HDContainer';
import HDInfoBox from './components/HDInfoBox/HDInfoBox';
import HDTickListItem from './components/HDTickListItem/HDTickListItem';
import * as yup from './yup';
import HDSwitch from './components/HDSwitch/HDSwitch';
import HDBubble from './components/HDBubble/HDBubble';
import HDCompletedCardInfo from './components/HDCompletedCardInfo/HDCompletedCardInfo';
import HDRibbon from './components/HDRibbon/HDRibbon';
import HDFullscreenLoader from './components/HDFullscreenLoader/HDFullscreenLoader';
import HDSpinner from './components/HDSpinner/HDSpinner';
import HDLoader from './components/HDLoader/HDLoader';
import HDCustomerReviews from './components/HDCustomerReviews/HDCustomerReviews';
import HDCustomerReviewsWidget from './components/HDCustomerReviews/HDCustomerReviewsWidget';
import HDPaymentBreakdown from './components/HDPaymentBreakdown/HDPaymentBreakdown';
import HDToast from './components/HDToast/HDToast';
import HDCheckbox from './components/HDCheckbox/HDCheckbox';
import HDOverlay from './components/HDOverlay/HDOverlay';
import HDPriceTable from './components/HDPriceTable/HDPriceTable';
import HDVehicleSimpleDetails from './components/HDVehicleDetails/HDVehicleSimpleDetails';
import HDRadioButtonList from './components/HDRadioButtonList/HDRadioButtonList';
import HDCheckboxButtonList from './components/HDCheckboxButtonList/HDCheckboxButtonList';
import HDPolicySelect from './components/HDPolicySelect/HDPolicySelect';

export {
    HDTextInput,
    HDTextInputRefactor,
    HDButton,
    HDButtonRefactor,
    HDLabel,
    HDLabelRefactor,
    HDInfoCard,
    HDInfoCardRefactor,
    HDToggleButtonGroup,
    HDToggleButtonGroupRefactor,
    HDClickableCard,
    HDDropdownList,
    HDInteractiveCard,
    HDInteractiveCardRefactor,
    HDImageRadioButton,
    HDButtonDashed,
    HDStepper,
    HDDataCard,
    HDForm,
    HDWarningModal,
    HDModal,
    HDAsyncSelect,
    HDOverlayPopup,
    HDInformationToolTip,
    HDAlert,
    HDAlertRefactor,
    HDDatePicker,
    HDDatePickerRefactor,
    HDTable,
    HDAccordion,
    HDAccordionRefactor,
    HDQuoteTable,
    HDQuoteInfo,
    HDQuoteInfoRefactor,
    HDQuoteInfoWarning,
    HDQuoteDownload,
    HDQuoteDownloadRefactor,
    yup,
    HDPlaceholderWithHeader,
    HDTickListItem,
    HDContainer,
    HDBubble,
    HDInfoBox,
    HDSwitch,
    HDLoader,
    HDCompletedCardInfo,
    HDRibbon,
    HDFullscreenLoader,
    HDSpinner,
    HDCustomerReviews,
    HDCustomerReviewsWidget,
    HDPaymentBreakdown,
    HDToast,
    HDCheckbox,
    HDOverlay,
    HDPriceTable,
    HDVehicleSimpleDetails,
    HDRadioButtonList,
    HDCheckboxButtonList,
    HDPolicySelect
};
