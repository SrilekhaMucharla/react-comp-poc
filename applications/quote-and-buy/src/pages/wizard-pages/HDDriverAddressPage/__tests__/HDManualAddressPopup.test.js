import React from 'react';
import { mount } from 'enzyme';
import { HastingsAddressLookupService } from 'hastings-capability-addresslookup';
import { HDButtonRefactor } from 'hastings-components';
import HDManualAddresssPopup from '../HDManualAddressPopup';
import createPortalRoot from '../../__helpers__/test/createPortalRoot';

describe('<HDManualAddresssPopup />', () => {
    createPortalRoot();

    let wrapper;

    const addressLookupMatches = [{
        matchAccuracy: 0.7,
        address: {
            matchAccuracy: 0.7,
            address: {
                addressLine1: '1 Westbourne Court',
                addressLine2: 'Cooden Drive',
                addressLine3: '',
                city: 'BEXHILL-ON-SEA',
                county: 'East Sussex',
                postalCode: 'TN393AA'
            }
        }
    }];
    jest.spyOn(HastingsAddressLookupService, 'lookupAddressByPostCode').mockResolvedValue({ result: { matches: addressLookupMatches } });

    beforeEach(() => {
        jest.clearAllTimers();
        jest.useFakeTimers();
        const address = {
            addressLine1: 'Test St 123',
            addressLine2: 'xxx',
            addressLine3: 'xxx',
            postalCode: 'XX23XX',
            city: 'Test City',
        };
        const handleConfirm = jest.fn();
        const trigger = (<div>TEST</div>);
        wrapper = mount(<HDManualAddresssPopup
            onConfirm={handleConfirm}
            initialAddress={address}
            trigger={trigger} />);
    });

    it('render component with all props', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('fill form and confirm', async () => {
        await act(async () => wrapper.find('HDOverlayPopup[id="manual-address-popup"] .hd-overlay-btn').invoke('onClick')());
        wrapper.update();

        let textField = wrapper.find('input').at(0);

        await act(async () => textField.props().onChange({
            target: {
                name: textField.prop('name'),
                value: 'Test Drive 100'
            }
        }));

        textField = wrapper.find('input').at(1);
        await act(async () => textField.props().onChange({
            target: {
                name: textField.prop('name'),
                value: 'test line2'
            }
        }));

        textField = wrapper.find('ForwardRef').at(2);
        await act(async () => textField.props().onChange({
            target: {
                name: textField.prop('name'),
                value: 'testline3'
            }
        }));

        textField = wrapper.find('input').at(3);
        await act(async () => textField.props().onChange({
            target: {
                name: textField.prop('name'),
                value: 'Test city'
            }
        }));

        wrapper.update();

        await act(async () => wrapper.find(HDButtonRefactor).at(0).props().onClick());
        wrapper.update();

        jest.runOnlyPendingTimers();
        wrapper.update();

        expect(wrapper.find('input')).toHaveLength(0);
    });
});
