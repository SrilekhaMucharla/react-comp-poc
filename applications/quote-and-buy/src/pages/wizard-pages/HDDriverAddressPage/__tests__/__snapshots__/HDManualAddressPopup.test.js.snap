// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDManualAddresssPopup /> render component with all props 1`] = `
<HDManualAddressPopup
  initialAddress={
    Object {
      "addressLine1": "Test St 123",
      "addressLine2": "xxx",
      "addressLine3": "xxx",
      "city": "Test City",
      "postalCode": "XX23XX",
    }
  }
  onConfirm={[MockFunction]}
  trigger={
    <div>
      TEST
    </div>
  }
>
  <ForwardRef
    cancelButton="Cancel"
    className="manual-address-popup"
    confirmButton="Update"
    disabledConfirmButton={false}
    id="manual-address-popup"
    onBeforeOpen={[Function]}
    onConfirm={[Function]}
    overlayButtonIcon={
      <div>
        TEST
      </div>
    }
    showButtons={true}
    webAnalyticsEvent={
      Object {
        "event_action": "Enter address manually",
      }
    }
    webAnalyticsView={
      Object {
        "page_section": "Enter address manually",
      }
    }
  >
    <ForwardRef
      cancelButton="Cancel"
      className="manual-address-popup"
      confirmButton="Update"
      disabledConfirmButton={false}
      id="manual-address-popup"
      onBeforeOpen={[Function]}
      onBlur={null}
      onCancel={null}
      onChange={null}
      onClick={null}
      onConfirm={[Function]}
      onDelete={null}
      onEdit={null}
      onFocus={null}
      onSelect={null}
      overlayButtonIcon={
        <div>
          TEST
        </div>
      }
      showButtons={true}
      webAnalyticsEvent={
        Object {
          "event_action": "Enter address manually",
        }
      }
    >
      <HDOverlayPopup
        cancelButton="Cancel"
        className="manual-address-popup"
        closeOnSelect={false}
        confirmButton="Update"
        customStyle=""
        disabledConfirmButton={false}
        headerBar={null}
        hidden={false}
        hideOnConfirm={true}
        id="manual-address-popup"
        labelText=""
        onBeforeClose={[Function]}
        onBeforeOpen={[Function]}
        onCancel={[Function]}
        onConfirm={[Function]}
        overlayButtonIcon={
          <div>
            TEST
          </div>
        }
        overlayButtonsClassName=""
        showButtons={true}
      >
        <div
          className="hd-information-tooltip manual-address-popup"
        >
          <div
            className="hd-overlay-btn "
            hidden={false}
            id="manual-address-popup"
            onClick={[Function]}
          >
            <div>
              TEST
            </div>
          </div>
        </div>
      </HDOverlayPopup>
    </ForwardRef>
  </ForwardRef>
</HDManualAddressPopup>
`;
